src/Bridge/Symfony/Bundle/Resources/config/metadata.xml

```diff
@@ -18,6 +18,16 @@
             <argument type="service" id="api_platform.metadata.resource.name_collection_factory.cached.inner" />
         </service>
 
+        <service id="api_platform.metadata.resource.name_collection_factory.yml" decorates="api_platform.metadata.resource.name_collection_factory" class="ApiPlatform\Core\Metadata\Resource\Factory\YmlResourceNameCollectionFactory" public="false">
```

Comment:
It's YAML, .yml is only the preferred file extension in projects such as Symfony.

`api_platform.metadata.resource.name_collection_factory.yaml`

`YamlResourceNameCollectionFactory`